src/kaltura-player.js

```diff
@@ -809,7 +808,7 @@ class KalturaPlayer extends FakeEventTarget {
             }
 
             if (typeof plugin.getEngineDecorator === 'function') {
-              registerEngineDecoratorProvider(plugin);
+              this._localPlayer.registerEngineDecoratorProvider(name, {getEngineDecorator: plugin.getEngineDecorator.bind(plugin)});
```

Comment:
Seems weird to write it that way.
Why not just pass the handler? and if you really need `getEngineDecorator` API why not to define `EngineDecoratorProvider` class which accepts the plugin and expose all needed fields ?
Something like:
```js
this._localPlayer.registerEngineDecoratorProvider(new EngineDecoratorProvider(plugin));

class EngineDecoratorProvider {
  constructor(plugin) {
    this.name = plugin.getName();
    this.hander = plugin.getEngineDecorator.bind(plugin);
  }

  getEngineDecorator() {
    return this.hander;
  }

  getName() {
    return this.name;
  }
}

```